dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGateway.scala

```diff
@@ -191,6 +191,21 @@ class AkkaHttpServiceGateway(
     headers.filterNot(header => HeadersToFilter(header.lowercaseName()))
   }
 
+  private def concatHeaders(
+      serviceResponseHeaders: immutable.Seq[HttpHeader],
+      webSocketClientResponseHeaders: immutable.Seq[HttpHeader]
+  ) = {
+    def addHeader(res: immutable.Seq[HttpHeader], header: HttpHeader) = {
+      if (res.exists(other => other.lowercaseName().equals(header.lowercaseName()))) {
+        res
+      } else {
+        res :+ header
+      }
+    }
+
+    filterHeaders(serviceResponseHeaders).foldLeft(webSocketClientResponseHeaders)(addHeader)
```

Comment:
The `filterHeaders` method is already filtering out a few related websocket headers the only problem is that it's not exhaustive. 

Maybe a better approach is to make it exhaustive. I find it a better approach because it's more explicit than `concatHeaders`. 

`concatHeaders` is concatenating and removing duplicates which also solves the problem.

To make it more explicit, we could rename `filterHeaders` to `filterOutWebSockerHeaders`

Reply:
Here are the list of headers I found at https://tools.ietf.org/html/rfc6455#page-57

* Sec-WebSocket-Key
* Sec-WebSocket-Extensions
* Sec-WebSocket-Accept
* Sec-WebSocket-Protocol
* Sec-WebSocket-Version

Reply:
I see your point. I'll make the change :)